src/js/hooks/cycleCount/useAllProductsTab.jsx

```diff
@@ -87,6 +89,44 @@ const useAllProductsTab = ({ filterParams }) => {
     filterParams,
   });
 
+  const {
+    selectRow,
+    isChecked,
+    selectHeaderCheckbox,
+    selectedRowsAmount,
+    headerCheckboxProps,
+  } = useTableCheckboxes();
+
+  const getProductIds = () => tableData.data.map((row) => row.product.id);
+
+  // Separated from columns to reduce the amount of rerenders of
+  // the rest columns (on checked checkboxes change)
+  const checkboxesColumn = columnHelper.accessor('selected', {
```

Comment:
do you think there would be any benefit from including the `useMemo` here not to re-render it every single time, since a lot of compoments are built here?

Reply:
There are benefits, because I can't use `useMemo` on all columns, because there are problems with refreshing the checkboxes (even with correct dependencies)